pyswarms/utils/search/base_search.py

```diff
@@ -13,6 +13,21 @@
 from past.builtins import xrange
 
 class SearchBase(object):
+
+    def assertions(self):
+        """Assertion method to check :code:`optimizer` input.
+
+        Raises
+        ------
+        TypeError
+            When :code:`optimizer` is not named LocalBestPSO or
+                GlobalBestPSO.
+        """
+        # Check type of optimizer object
+        if self.optimizer.__name__ not in ('LocalBestPSO', 'GlobalBestPSO'):
+            raise TypeError('Parameter `optimizer` must be of type '
+                            'LocalBestPSO or GlobalBestPSO')
+
```

Comment:
This is good for now and I can accept this in its current form. 😄 👍 

But I'm wondering how can we scale this when more search methods will be added. For example, Binary PSO is a single-objective method, and I think the same search methods are applicable to it. 

The same will go perhaps in multi-objective and constrained optimization. Since the search methods are "independent" whatever optimization method we are using. Meaning, it doesn't really care what kind of problem we are trying to solve, as long as we have an `optimize` and `optimizer` supplied as params. Then it's possible to apply the same search methods for these kind of techniques in the future. What do you think @SioKCronin?

I'm thinking of instead checking if the object passed in the `optimizer`  argument belongs to the `pyswarms.base` module. __OR__ check if the object passed in the `optimizer` method has an `optimize` attribute (thus using a `hasattr` as a checker). What do you think @jazcap53?

Reply:
Hi @ljvmiranda921,
I'll of course make the changes you suggest with pleasure. :smile:
> I'd appreciate it if you can rebase your commit similar to the title of your pull request  

I had trouble getting git to accept a commit title with spaces in it. I'll try using `\<space>` instead of `<space>`, and if that doesn't work I'll try using `%20`.

> I'm wondering how can we scale this when more search methods will be added

I was wondering the same thing. I had trouble implementing it, though, since the `optimizer` parameter is a class rather than an instance -- I'll research it further and ask help if I can't find a solution.

> Is a parenthesis missing here?

Sure enough. I was afraid to change that line, though -- it's not mine. I just appears in the diff because my vim is set to eliminate trailing spaces on save of a .py file. I'll rephrase it according to your suggestion.

I'm enjoying this very much...  :+1:

Reply:
@ljvmiranda921   Figured out that I have to use `issublcass()`. Simple (embarrassing). :smile:

Reply:
@ljvmiranda921    o.i.c.   `if not hasattr(self.optimizer, 'optimize'):`
I'll use that unless you would prefer something else  :smile: